recipes/cpp-ipc/all/conanfile.py

```diff
@@ -21,64 +29,67 @@ class CppIPCConan(ConanFile):
         "fPIC": True,
     }
 
-    generators = "cmake"
-
-    _compiler_required_cpp17 = {
-        "Visual Studio": "17",
-        "gcc": "8",
-        "clang": "4",
-    }
-
-    _cmake = None
+    @property
+    def _min_cppstd(self):
+        return 17
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8",
+            "clang": "4",
+            "msvc": "193",
+            "Visual Studio": "17",
+        }
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def validate(self):
-        if tools.is_apple_os(self.settings.os):
-            raise ConanInvalidConfiguration("{} does not support Apple platform".format(self.name))
+        if self.settings.os == "Macos":
```

Comment:
Why not use [conan.tools.apple.is_apple_os](https://docs.conan.io/2/reference/tools/apple/other.html?highlight=is_apple_os#conan.tools.apple.is_apple_os)?

Reply:
No real reason, I mistakenly assumed the function has been deprecated in v2. Thanks for mentioning!

Reply:
Yeah, I figured it wouldn't work on iOS etc, too. Thanks for updating this!